src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1199,14 +1195,10 @@ int sd_dhcp_server_add_vendor_option(sd_dhcp_server *server, sd_dhcp_option *v)
         assert_return(server, -EINVAL);
         assert_return(v, -EINVAL);
 
-        r = ordered_hashmap_ensure_allocated(&server->vendor_options, &dhcp_option_hash_ops);
+        r = ordered_hashmap_ensure_put(&server->vendor_options, &dhcp_option_hash_ops, v, v);
```

Comment:
Hmm, this is not a fault in this PR, but we should use `ordered_set` instead of `ordered_hashmap` here.